custom-vu/src/main/kotlin/jces1209/vu/page/filters/ServerFiltersPage.kt

```diff
@@ -0,0 +1,25 @@
+package jces1209.vu.page.filters
+
+import com.atlassian.performance.tools.jiraactions.api.WebJira
+import jces1209.vu.wait
+import org.openqa.selenium.By
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfElementLocated
+
+class ServerFiltersPage(
+    private val jira: WebJira,
+    private val driver: WebDriver
+) : FiltersPage {
+
+    private val table = By.id("mf_popular")
+
+    override fun open(): FiltersPage {
```

Comment:
Just a question out of curiosity would it make sense to use here kotlin apply, as:
`override fun open(): FiltersPage = apply { jira.navigateTo("secure/ManageFilters.jspa?filterView=popular") }` ?

Reply:
It would result in the same behavior, yes. But it doesn't have to `return this`, it could return something else, e.g. navigate, read some elements and put them in another `FiltersPage`.
The [`apply`](https://kotlinlang.org/docs/reference/scope-functions.html#apply) seems to communicate side-effects.
TBH this method is a bit side-effecty on the interface level. E.g. `listFilters` depends on the hidden state set by `open`, but the interface allows to call `listFilters` without ever calling `open`.
A stronger design would move `open` away from the `FiltersPage` (how could a page know how to open itself?) and move it to something like `FiltersFeature`.

I didn't add the appropriate design here, because I took shortcuts to meet the todays deadline 😬

Reply:
Thx for explanation :)